File: estonian-courier-sim/app/src/main/java/ee/kuller/app/ui/screens/HomeScreen.kt
package ee.kuller.app.ui.screens

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.LocationOn
import androidx.compose.material.icons.filled.Place
import androidx.compose.material3.Button
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Switch
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import ee.kuller.app.GameViewModel
import ee.kuller.app.data.Content
import ee.kuller.app.model.Order
import ee.kuller.app.ui.StatPill

@Composable
fun HomeScreen(vm: GameViewModel, onAccept: (Order) -> Unit) {
    LazyColumn(
        Modifier.fillMaxSize().padding(horizontal = 16.dp),
        verticalArrangement = Arrangement.spacedBy(10.dp)
    ) {
        item {
            Text(
                "Tere, kuller! 🛵",
                fontWeight = FontWeight.Bold,
                fontSize = 26.sp,
                color = MaterialTheme.colorScheme.onBackground,
                modifier = Modifier.padding(top = 16.dp)
            )
            Text(
                "Учи эстонский, развозя заказы",
                fontSize = 14.sp,
                color = MaterialTheme.colorScheme.onBackground.copy(alpha = 0.7f)
            )
        }

        item {
            Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.spacedBy(10.dp)) {
                StatPill("💶", vm.moneyStr(), "Заработок", Modifier.weight(1f))
                StatPill("⭐", vm.ratingStr(), "Рейтинг", Modifier.weight(1f))
                StatPill("📦", "${vm.state.deliveries}", "Доставки", Modifier.weight(1f))
                StatPill("🎚", "Lvl ${vm.state.level}", "Уровень", Modifier.weight(1f))
            }
        }

        item { OnlineCard(vm) }

        if (vm.online) {
            item {
                Text(
                    "Доступные заказы",
                    fontWeight = FontWeight.Bold,
                    fontSize = 18.sp,
                    color = MaterialTheme.colorScheme.onBackground,
                    modifier = Modifier.padding(top = 8.dp)
                )
            }
            items(Content.orders) { order ->
                OrderCard(order, onAccept = { onAccept(order) })
            }
            item { Text("") }
        }
    }
}

@Composable
private fun OnlineCard(vm: GameViewModel) {
    Card(
        shape = RoundedCornerShape(20.dp),
        colors = CardDefaults.cardColors(
            containerColor = if (vm.online) MaterialTheme.colorScheme.primaryContainer
            else MaterialTheme.colorScheme.surface
        )
    ) {
        Row(
            Modifier.fillMaxWidth().padding(16.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Column(Modifier.weight(1f)) {
                Text(
                    if (vm.online) "Вы на линии" else "Вы офлайн",
                    fontWeight = FontWeight.Bold,
                    fontSize = 18.sp,
                    color = MaterialTheme.colorScheme.onSurface
                )
                Text(
                    if (vm.online) "Принимайте заказы и учите слова"
                    else "Включите смену, чтобы получать заказы",
                    fontSize = 13.sp,
                    color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.7f)
                )
            }
            Switch(checked = vm.online, onCheckedChange = { vm.toggleOnline() })
        }
    }
}

@Composable
private fun OrderCard(order: Order, onAccept: () -> Unit) {
    Card(
        Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(20.dp),
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
        elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
    ) {
        Column(Modifier.padding(16.dp)) {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Text("🍽", fontSize = 22.sp)
                Column(Modifier.weight(1f).padding(start = 8.dp)) {
                    Text(order.restaurant, fontWeight = FontWeight.Bold, fontSize = 18.sp,
                        color = MaterialTheme.colorScheme.onSurface)
                    Text(order.itemsRu, fontSize = 13.sp,
                        color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.7f))
                }
                Text(
                    "+%.2f €".format(order.payout),
                    fontWeight = FontWeight.Bold,
                    fontSize = 18.sp,
                    color = MaterialTheme.colorScheme.primary
                )
            }
            Row(
                Modifier.padding(top = 10.dp),
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(4.dp)
            ) {
                Icon(Icons.Filled.Place, null, Modifier.size(16.dp),
                    tint = MaterialTheme.colorScheme.secondary)
                Text(order.address, fontSize = 13.sp,
                    color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.8f))
                Text(" · %.1f km".format(order.distanceKm), fontSize = 13.sp,
                    color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.6f))
            }
            Button(
                onClick = onAccept,
                modifier = Modifier.fillMaxWidth().padding(top = 12.dp),
                shape = RoundedCornerShape(14.dp)
            ) {
                Icon(Icons.Filled.LocationOn, null, Modifier.size(18.dp))
                Text("  Принять заказ", fontWeight = FontWeight.Bold)
            }
        }
    }
}

File: kuller_flutter/lib/screens/home_screen.dart
import 'package:flutter/material.dart';

import '../game_vm.dart';
import '../models.dart';
import '../widgets.dart';

class HomeScreen extends StatelessWidget {
  final GameViewModel vm;
  final void Function(Order) onAccept;
  const HomeScreen({super.key, required this.vm, required this.onAccept});

  @override
  Widget build(BuildContext context) {
    final scheme = Theme.of(context).colorScheme;
    return ListView(
      padding: const EdgeInsets.symmetric(horizontal: 16),
      children: [
        Padding(
          padding: const EdgeInsets.only(top: 16),
          child: Text('Tere, kuller! 🛵',
              style: TextStyle(
                  fontWeight: FontWeight.bold,
                  fontSize: 26,
                  color: scheme.onSurface)),
        ),
        Text('Учи эстонский, развозя заказы',
            style: TextStyle(
                fontSize: 14, color: scheme.onSurface.withOpacity(0.7))),
        const SizedBox(height: 10),
        Row(
          children: [
            Expanded(child: StatPill('💶', vm.moneyStr(), 'Заработок')),
            const SizedBox(width: 10),
            Expanded(child: StatPill('⭐', vm.ratingStr(), 'Рейтинг')),
            const SizedBox(width: 10),
            Expanded(child: StatPill('📦', '${vm.state.deliveries}', 'Доставки')),
            const SizedBox(width: 10),
            Expanded(child: StatPill('🎚', 'Lvl ${vm.state.level}', 'Уровень')),
          ],
        ),
        const SizedBox(height: 10),
        _OnlineCard(vm: vm),
        if (vm.online) ...[
          Padding(
            padding: const EdgeInsets.only(top: 18),
            child: Row(
              children: [
                Expanded(
                  child: Text('Доступные заказы',
                      style: TextStyle(
                          fontWeight: FontWeight.bold,
                          fontSize: 18,
                          color: scheme.onSurface)),
                ),
                OutlinedButton.icon(
                  onPressed: vm.refreshOrders,
                  icon: const Icon(Icons.refresh, size: 16),
                  label: const Text('Обновить'),
                ),
              ],
            ),
          ),
          const SizedBox(height: 6),
          for (final order in vm.availableOrders)
            _OrderCard(order: order, onAccept: () => onAccept(order)),
          const SizedBox(height: 24),
        ],
      ],
    );
  }
}

class _OnlineCard extends StatelessWidget {
  final GameViewModel vm;
  const _OnlineCard({required this.vm});

  @override
  Widget build(BuildContext context) {
    final scheme = Theme.of(context).colorScheme;
    return Card(
      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(20)),
      color: vm.online ? scheme.primaryContainer : scheme.surface,
      child: Padding(
        padding: const EdgeInsets.all(16),
        child: Row(
          children: [
            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text(vm.online ? 'Вы на линии' : 'Вы офлайн',
                      style: TextStyle(
                          fontWeight: FontWeight.bold,
                          fontSize: 18,
                          color: scheme.onSurface)),
                  Text(
                      vm.online
                          ? 'Принимайте заказы и учите слова'
                          : 'Включите смену, чтобы получать заказы',
                      style: TextStyle(
                          fontSize: 13,
                          color: scheme.onSurface.withOpacity(0.7))),
                ],
              ),
            ),
            Switch(value: vm.online, onChanged: (_) => vm.toggleOnline()),
          ],
        ),
      ),
    );
  }
}

class _OrderCard extends StatelessWidget {
  final Order order;
  final VoidCallback onAccept;
  const _OrderCard({required this.order, required this.onAccept});

  @override
  Widget build(BuildContext context) {
    final scheme = Theme.of(context).colorScheme;
    return Card(
      elevation: 2,
      margin: const EdgeInsets.symmetric(vertical: 5),
      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(20)),
      color: scheme.surface,
      child: Padding(
        padding: const EdgeInsets.all(16),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Row(
              children: [
                const Text('🍽', style: TextStyle(fontSize: 22)),
                const SizedBox(width: 8),
                Expanded(
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Text(order.restaurant,
                          style: TextStyle(
                              fontWeight: FontWeight.bold,
                              fontSize: 18,
                              color: scheme.onSurface)),
                      Text(order.itemsRu,
                          style: TextStyle(
                              fontSize: 13,
                              color: scheme.onSurface.withOpacity(0.7))),
                    ],
                  ),
                ),
                Text('+${order.payout.toStringAsFixed(2)} €',
                    style: TextStyle(
                        fontWeight: FontWeight.bold,
                        fontSize: 18,
                        color: scheme.primary)),
              ],
            ),
            Padding(
              padding: const EdgeInsets.only(top: 10),
              child: Row(
                children: [
                  Icon(Icons.place, size: 16, color: scheme.secondary),
                  const SizedBox(width: 4),
                  Flexible(
                    child: Text(order.address,
                        style: TextStyle(
                            fontSize: 13,
                            color: scheme.onSurface.withOpacity(0.8))),
                  ),
                  Text(' · ${order.distanceKm.toStringAsFixed(1)} km',
                      style: TextStyle(
                          fontSize: 13,
                          color: scheme.onSurface.withOpacity(0.6))),
                ],
              ),
            ),
            Padding(
              padding: const EdgeInsets.only(top: 12),
              child: SizedBox(
                width: double.infinity,
                child: FilledButton.icon(
                  onPressed: onAccept,
                  icon: const Icon(Icons.location_on, size: 18),
                  label: const Text('Принять заказ',
                      style: TextStyle(fontWeight: FontWeight.bold)),
                  style: FilledButton.styleFrom(
                    shape: RoundedRectangleBorder(
                        borderRadius: BorderRadius.circular(14)),
                  ),
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}
